src/tools/cephfs/first-damage.py

```diff
@@ -76,11 +79,19 @@ def traverse(MEMO, ioctx):
             it = ioctx.get_omap_vals(rctx, None, None, 100000)[0]
             ioctx.operate_read_op(rctx, o.key)
             for (dnk, val) in it:
-                log.debug('\t%s', dnk)
+                log.debug('\t%s: val size %d', dnk, len(val))
                 (first,) = struct.unpack('<I', val[:4])
                 if first > NEXT_SNAP:
                     log.warning(f"found {o.key}:{dnk} first (0x{first:x}) > NEXT_SNAP (0x{NEXT_SNAP:x})")
-                    if REMOVE:
+                    if REPAIR_NOSNAP and dnk.endswith("_head") and first == CEPH_NOSNAP:
+                        log.warning(f"repairing first==CEPH_NOSNAP damage, setting to NEXT_SNAP (0x{NEXT_SNAP:x})")
+                        first = NEXT_SNAP
+                        nval = bytearray(val)
+                        struct.pack_into("<I", nval, 0, NEXT_SNAP)
```

Comment:
This looks right.

@batrick Help me remind, we never tested in teuthology, isn't it? (although the test script is under qa/workunits).

Reply:
No, we don't. It's just a script for QE to run.